rcl/src/rcl/subscription.c

```diff
@@ -193,6 +210,181 @@ rcl_subscription_get_default_options()
+}
+
+rcl_subscription_content_filter_options_t
+rcl_get_zero_initialized_subscription_content_filter_options()
+{
+  return (const rcl_subscription_content_filter_options_t) {
+           .allocator = rcl_get_default_allocator(),
+           .rmw_subscription_content_filter_options =
+             rmw_get_zero_initialized_content_filter_options()
+  };  // NOLINT(readability/braces): false positive
+}
+
+rcl_ret_t
+rcl_subscription_content_filter_options_init(
+  const char * filter_expression,
+  size_t expression_parameters_argc,
+  const char * expression_parameter_argv[],
+  rcl_subscription_content_filter_options_t * options)
+{
+  RCL_CHECK_ARGUMENT_FOR_NULL(options, RCL_RET_INVALID_ARGUMENT);
+  const rcl_allocator_t * allocator = &options->allocator;
+  RCL_CHECK_ALLOCATOR_WITH_MSG(allocator, "invalid allocator", return RCL_RET_INVALID_ARGUMENT);
+
+  rmw_ret_t rmw_ret = rmw_subscription_content_filter_options_init(
+    filter_expression,
+    expression_parameters_argc,
+    expression_parameter_argv,
+    allocator,
+    &options->rmw_subscription_content_filter_options
+  );
+
+  return rcl_convert_rmw_ret_to_rcl_ret(rmw_ret);
+}
+
+rcl_ret_t
+rcl_subscription_content_filter_options_set(
+  const char * filter_expression,
+  size_t expression_parameters_argc,
+  const char * expression_parameter_argv[],
+  rcl_subscription_content_filter_options_t * options)
+{
+  RCL_CHECK_ARGUMENT_FOR_NULL(filter_expression, RCL_RET_INVALID_ARGUMENT);
+  RCL_CHECK_ARGUMENT_FOR_NULL(options, RCL_RET_INVALID_ARGUMENT);
+  const rcl_allocator_t * allocator = &options->allocator;
+  RCL_CHECK_ALLOCATOR_WITH_MSG(allocator, "invalid allocator", return RCL_RET_INVALID_ARGUMENT);
+
+  rmw_ret_t ret = rmw_subscription_content_filter_options_set(
+    filter_expression,
+    expression_parameters_argc,
+    expression_parameter_argv,
+    allocator,
+    &options->rmw_subscription_content_filter_options
+  );
+  return rcl_convert_rmw_ret_to_rcl_ret(ret);
+}
+
+rcl_ret_t
+rcl_subscription_content_filter_options_fini(
+  rcl_subscription_content_filter_options_t * options)
+{
+  RCL_CHECK_ARGUMENT_FOR_NULL(options, RCL_RET_INVALID_ARGUMENT);
+  const rcl_allocator_t * allocator = &options->allocator;
+  RCL_CHECK_ALLOCATOR_WITH_MSG(allocator, "invalid allocator", return RCL_RET_INVALID_ARGUMENT);
+
+  rmw_ret_t ret = rmw_subscription_content_filter_options_fini(
+    &options->rmw_subscription_content_filter_options,
+    allocator
+  );
+
+  return rcl_convert_rmw_ret_to_rcl_ret(ret);
+}
+
+rcl_ret_t
+rcl_subscription_set_content_filter(
+  const rcl_subscription_t * subscription,
+  const rcl_subscription_content_filter_options_t * options
+)
+{
+  (void)subscription;
+  (void)options;
+
+  return RCL_RET_UNSUPPORTED;
+}
+
+rcl_ret_t
+rcl_subscription_get_content_filter(
+  const rcl_subscription_t * subscription,
+  rcl_subscription_content_filter_options_t * options
+)
+{
+  (void)subscription;
+  (void)options;
+
+  return RCL_RET_UNSUPPORTED;
```

Comment:
why these are not supported? is it supposed to call `rmw_subscription_get_content_filter`?